src/nodal_properties.cc

```diff
@@ -0,0 +1,30 @@
+#include "nodal_properties.h"
+
+// Function to create new property with given name and size (rows x cols)
+void mpm::NodalProperties::create_property(const std::string& property,
+                                           unsigned rows, unsigned columns) {
+  // Initialize a matrix with size of rows times columns and insert it to the
+  // property database map
+  Eigen::MatrixXd property_data = Eigen::MatrixXd::Zero(rows, columns);
+  properties_.insert(
+      std::pair<std::string, Eigen::MatrixXd>(property, property_data));
+}
+
+// Return data in the nodal properties map at a specific index
+Eigen::MatrixXd mpm::NodalProperties::property(const std::string& property,
+                                               unsigned node_id,
+                                               unsigned mat_id,
+                                               unsigned nprops) const {
+  double* position = &properties_.at(property)(node_id * nprops, mat_id);
```

Comment:
```suggestion
  const double* position = &properties_.at(property)(node_id * nprops, mat_id);
```

Reply:
The return type here is `const double *` and not `double *`, hence the failure in clang